sycl/source/detail/graph_impl.hpp

```diff
@@ -313,6 +340,16 @@ class graph_impl {
     MInorderQueueMap[QueueWeakPtr] = Node;
   }
 
+  /// Prints contents of graph to text file in DOT format
+  /// \param GraphName is a string appended to the output file name
+  void printGraphAsDot(const char *GraphName);
+
+  /// Checks is the graph_impl of Graph has a similar structure to
+  /// the graph_impl of the caller.
+  /// \param Graph is reference to the graph to compare with.
+  bool hasSimilarStructure(std::shared_ptr<detail::graph_impl> Graph,
+                           int DebugPrint);
+
```

Comment:
Both these methods are also relevant to executable graphs. Would like `exec_graph_impl` to have equivalent functionality for printing and comparison.